contrib/mesos/docs/ha.md

```diff
@@ -66,5 +86,417 @@ The command line parameters that affect the hash calculation are listed below.
+    imagePullPolicy: IfNotPresent
+    command:
+    - /opt/kubernetes/km
+    - controller-manager
+    - --master=http://${NGINX_IP}:${NGINX_APISERVER_PORT}
+    - --cloud-provider=mesos
+    - --cloud-config=/etc/kubernetes/mesos-cloud.conf
+    volumeMounts:
+    - mountPath: /etc/kubernetes
+      name: kubernetes-config
+      readOnly: true
+  volumes:
+  - hostPath:
+      path: /etc/kubernetes
+    name: kubernetes-config
+EOF
+```
+
+Controller manager also needs a mesos configuration file as one of its parameters, and this
+configuration file is written to `/etc/kubernetes/mesos-cloud.conf`
+
+```shell
+$ cat <<EOF >/etc/kubernetes/mesos-cloud.conf
+[mesos-cloud]
+        mesos-master        = ${MESOS_IP}:${MESOS_PORT}
+EOF
+```
+
+### Configure Apiserver
+
+Apiserver runs on every master node, so its specification file is put into
+`/etc/kubernetes/manifests/`
+
+```shell
+cat <<EOF > /etc/kubernetes/manifests/apiserver.yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: kube-apiserver
+  namespace: kube-system
+spec:
+  hostNetwork: true
+  containers:
+  - name: kube-apiserver
+    image: haih/k8sm:latest
+    imagePullPolicy: IfNotPresent
+    command:
+    - /opt/kubernetes/km
+    - apiserver
+    - --insecure-bind-address=0.0.0.0
+    - --etcd-servers=http://${ETCD_IP}:${ETCD_PORT}
+    - --allow-privileged=true
+    - --service-cluster-ip-range=10.10.10.0/24
+    - --insecure-port=${K8S_APISERVER_PORT}
+    - --cloud-provider=mesos
+    - --cloud-config=/etc/kubernetes/mesos-cloud.conf
+    - --advertise-address=${MY_IP}
+    ports:
+    - containerPort: ${K8S_APISERVER_PORT}
+      hostPort: ${K8S_APISERVER_PORT}
+      name: local
+    volumeMounts:
+    - mountPath: /etc/kubernetes
+      name: kubernetes-config
+      readOnly: true
+  volumes:
+  - hostPath:
+      path: /etc/kubernetes
+    name: kubernetes-config
+EOF
+```
+
+Again, one would need to change `$MY_IP` to either `$K8S_1_IP` or `K8S_2_IP`
+depending on which master node one is currently working on.
+
+To summarize our current setup: we have apiserver and podmaster's pod
+specification files put into `/etc/kubernetes/manifests/` so they run on every
+master node.  Scheduler and controller manager's pod specification files are
+put into `/srv/kubernetes/manifests/`, and they will be copied into
+`/etc/kubernetes/manifests/` if and only if its podmaster was elected the
+leader.
+
+### Configure Nginx
+
+Nginx needs to be configured to load balance for both the apiserver and scheduler.
+For testing purpose, one can start Nginx in a Docker container
+
+```shell
+cat <<EOF >nginx.conf
+events {
+  worker_connections  4096;  ## Default: 1024
+}
+
+http {
+  upstream apiservers {
+    server ${K8S_1_IP}:${K8S_APISERVER_PORT};
+    server ${K8S_2_IP}:${K8S_APISERVER_PORT};
+  }
+
+  upstream schedulers {
+    server ${K8S_1_IP}:${K8S_SCHEDULER_PORT};
+    server ${K8S_2_IP}:${K8S_SCHEDULER_PORT};
+  }
+
+  server {
+    listen ${NGINX_APISERVER_PORT};
```

Comment:
Can you give me a pointer on which kubectl commands will not work without TLS/SPDY?